File: #Leetcode(Python)/Palindrome_Number.py
# Question:-

# Given an integer x, return true if x is a palindrome, and false otherwise.

# Solution:-

class Solution:
    def isPalindrome(self, x: int) -> bool:
        string_x = str(x)
        rev_string_x = string_x[::-1]
        if (string_x == rev_string_x):
            return True
        else:
            return False
            
        
t = int(input("Enter no. of test cases: "))
for i in range (0,t):
    x = int(input("Enter the number: "))

    solution = Solution()
    result = solution.isPalindrome(x)
    print(result)

File: #Leetcode(C++)/Palindrome_Number.cpp
// Question:-

// Given an integer x, return true if x is a palindrome, and false otherwise.

// Solution:-

#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        long long RevNum = 0;
        int Duplicate = x;
        while (x > 0){
            int ld = x % 10;
            RevNum = (RevNum * 10 ) + ld;
            x = x / 10;
            
        }
        
        if(RevNum == Duplicate){
            return true;
        }
        else{
            return false;
        }
    }
};

int main(){
    int t;
    cin >> t;
    for (int i = 0; i<t;i++){
        int x;
        cout << "Enter a number: ";
        cin >> x;

        Solution solution;
        int PalinNum = solution.isPalindrome(x);
        cout << PalinNum << endl;
    }
    return 0;   
}